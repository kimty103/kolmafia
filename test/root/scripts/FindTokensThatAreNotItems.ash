void process(coinmaster cm, string token) {
	if (token != "") {
		string out = to_string(cm) + " uses " + token;
		item tokenItem = to_item(token);
		if (tokenItem == $item[none]) {
			out = to_string(cm) + " uses " + token + " which is not an item.";
			print(out);
		}		
	}
}
foreach cm in $coinmasters[]
{
	string token = cm.token;
	process(cm, token);
	string clist = cm.multiple_tokens;
  	string[] tokens = split_string(clist, ",");
	foreach it in tokens {
		process(cm, tokens[it]);
	}
}
print("Coinmaster checks all done.")
