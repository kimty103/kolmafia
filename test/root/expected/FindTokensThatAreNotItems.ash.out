Crimbo 2011 uses Candy Credit which is not an item.
Dimemaster uses dime which is not an item.
PirateRealm Fun-a-Log uses FunPoint which is not an item.
Game Shoppe uses store credit which is not an item.
The Pokémporium uses pokédollar bills which is not an item.
The Swagger Shop uses swagger which is not an item.
Coinmaster checks all done.